gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -301,20 +316,24 @@ private WriteObjectResponse doResumableUpload() throws IOException {
                   ? responseObserver.transientError
                   : responseObserver.nonTransientError);
           break;
-        } else if (objectFinalized) {
+        }
+        if (objectFinalized
```

Comment:
Please add comment explaining WHY we are making this change. It is not very obvious from code why we are doing this.

Reply:
Enhanced logging.

Reply:
Most of the comments are covering WHAT we are doing. WHY we are doing the way we are is not very evident. Can we also add comments explaining the WHY?